concourse/client/util.py

```diff
@@ -122,3 +129,73 @@ def jobs_not_triggered(
             for pr_resource_version in pr_resource_versions:
                 if not wait_for_job_to_be_triggered(job, pr_resource_version, concourse_api):
                     yield (job, pr_resource, pr_resource_version)
+
+
+def process_untriggered_job(
+    job: Job,
+    resource: PipelineConfigResource,
+    resource_version: ResourceVersion,
+    concourse_api,
+):
+    cc_resource = concourse_api.resource(
+        pipeline_name=resource.pipeline_name(),
+        resource_name=resource.name,
+    )
+
+    if not cc_resource.is_pinned():
+        handle_resource_pinning(
+            job=job,
+            resource=cc_resource,
+            resource_version=resource_version,
+            concourse_api=concourse_api,
+        )
+    else:
+        # to be implemeted
+        pass
+
+
+def handle_resource_pinning(
+    job: Job,
+    resource: PipelineResource,
+    resource_version: ResourceVersion,
+    concourse_api,
+):
+    pin_comment = {
+        'datetime': datetime.now(),
```

Comment:
it would be easier to read (for humans..) if you formatted this to an isocalendar date